src/codegen/compvars.cpp

```diff
@@ -330,6 +330,57 @@ class UnknownType : public ConcreteCompilerType {
         return new ConcreteCompilerVariable(UNKNOWN, rtn, true);
     }
 
+    CompilerVariable* getiter(IREmitter& emitter, const OpInfo& info, ConcreteCompilerVariable* var) override {
+        static const std::string hasnext_str("__hasnext__");
+        static const std::string iter_str = "__iter__";
+
+        llvm::AllocaInst* rtn_value_memory = emitter.getBuilder()->CreateAlloca(g.llvm_value_type_ptr);
+
+        CallattrFlags flags = {.cls_only = true, .null_on_nonexistent = true };
+        CompilerVariable* iter_call = var->callattr(emitter, info, &iter_str, flags, ArgPassSpec(0), {}, 0);
```

Comment:
What do you think about adding something like a `hasattr` CompilerType method (returning yes/no/maybe)?  This could let us skip the dynamic checks for known types.  This could also replace the `hasnext != UNKNOWN && hasnext != UNDEF` -- I think that's equivalent to `hasattr() == YES` but the hasattr version feels more reliable and less likely to break subtly.

Also if we wanted, we could then simplify this function to
```
if (this->hasattr("__iter__") == YES && this->getattrType("__iter__")->callType()->hasattr("__hasnext__") == YES)
    return this->call("__iter__");
return emitter.call(g.funcs.getiter, value);
```
which I think still handles the common cases, though I'm not sure how it otherwise compares to what you've already written.